flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/DescribeFunctionOperation.java

```diff
@@ -0,0 +1,129 @@
+        params.put("identifier", sqlIdentifier);
+        params.put("isExtended", isExtended);
+        return OperationUtils.formatWithChildren(
+                "DESCRIBE FUNCTION", params, Collections.emptyList(), Operation::asSummaryString);
+    }
+
+    @Override
+    public TableResultInternal execute(Context ctx) {
+        // DESCRIBE FUNCTION <function> shows all the function properties.
+        Optional<ContextResolvedFunction> functionOpt =
+                ctx.getFunctionCatalog().lookupFunction(UnresolvedIdentifier.of(sqlIdentifier));
+        if (!functionOpt.isPresent()) {
+            throw new ValidationException(
+                    String.format(
+                            "Function with the identifier '%s' doesn't exist.",
+                            sqlIdentifier.asSummaryString()));
+        }
+        ContextResolvedFunction function = functionOpt.get();
+        CatalogFunction catalogFunction = function.getCatalogFunction();
+
+        if (catalogFunction == null && !isExtended) {
+            throw new ValidationException(
+                    String.format(
+                            "Function with the identifier '%s' is a system function which can only be described by using the extended keyword. Use 'DESCRIBE FUNCTION EXTENDED %s' to see its properties.",
+                            sqlIdentifier.asSummaryString(), sqlIdentifier.asSummaryString()));
+        }
+
+        List<List<Object>> rows = new ArrayList<>();
+        if (catalogFunction != null) {
+            rows.add(Arrays.asList("class name", catalogFunction.getClassName()));
+            rows.add(
+                    Arrays.asList(
+                            "function language", catalogFunction.getFunctionLanguage().toString()));
+            rows.add(
+                    Arrays.asList(
+                            "resource uris", catalogFunction.getFunctionResources().toString()));
+            rows.add(Arrays.asList("temporary", function.isTemporary()));
+        }
+
+        if (isExtended) {
+            FunctionDefinition definition = function.getDefinition();
+            rows.add(Arrays.asList("kind", definition.getKind().toString()));
+            rows.add(Arrays.asList("requirements", definition.getRequirements().toString()));
+            rows.add(Arrays.asList("deterministic", definition.isDeterministic()));
+            rows.add(Arrays.asList("constant folding", definition.supportsConstantFolding()));
+            rows.add(
+                    Arrays.asList(
+                            "signature",
+                            generateSignature(
+                                    definition.getTypeInference(
+                                            ctx.getCatalogManager().getDataTypeFactory()),
+                                    function.toString(),
+                                    definition)));
+        }
+
+        return buildTableResult(
+                new String[] {"info name", "info value"},
```

Comment:
Nit: Maybe the columns can be just `name`, `value`? Info sounds a bit informal, but `metadata` sounds a bit much too.

Reply:
I agree that info seems a bit redundant and informal, but this is consistent with what `DESCRIBE CATALOG` currently does https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/describe/#describe-statements. I would be open to changing it, but preferably to change both to keep them consistent. Snowflake for example does `property` and `value` https://docs.snowflake.com/en/sql-reference/sql/desc-function

Reply:
I see.  Ok, works for me.